nucypher/blockchain/eth/actors.py

```diff
@@ -112,6 +113,10 @@ def __repr__(self):
         r = r.format(class_name, self.checksum_address)
         return r
 
+    def __eq__(self, other) -> bool:
+        """Actors are equal if they have the same address."""
+        return bool(self.checksum_address == other.checksum_address)
```

Comment:
This will raise `AttributeError` is the other doesn't have a `checksum_address`.  This might be OK, but also maybe we want to catch it and raise `TypeError` instead?

Reply:
My initial impression is that returning `False` is desirable is that situation.